### PR TITLE
Add more in-depth container customization instructions

### PR DESCRIPTION
**Pull request by sdake**
**What this PR does / why we need it**:
The instructions aren't quite right after the mono-repo merge. This
patch fixes that problem. This PR also adds a -a to updateVersions.sh
to allow a global replacement of all container hub:tags.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```NONE
```

### CODE REVIEW

File: install/updateVersion.sh [hunk 1]
```diff
@@ -64,6 +66,15 @@ while getopts :gi:n:p:x:c:r:sA:P:E: arg; do
   esac
 done
 
+if [[ -n ${ALL_HUB_TAG} ]]; then
+    PILOT_HUB="$(echo ${ALL_HUB_TAG}|cut -f1 -d,)"
+    PILOT_TAG="$(echo ${ALL_HUB_TAG}|cut -f1 -d,)"
```
Comment (baodongli) [inline install/updateVersion.sh]: should be -f2?

File: install/updateVersion.sh [hunk 2]
```diff
@@ -64,6 +66,15 @@ while getopts :gi:n:p:x:c:r:sA:P:E: arg; do
...
+    MIXER_HUB="$(echo ${ALL_HUB_TAG}|cut -f1 -d,)"
+    MIXER_TAG="$(echo ${ALL_HUB_TAG}|cut -f1 -d,)"
```
Comment (baodongli) [inline install/updateVersion.sh]: ditto

File: install/updateVersion.sh [hunk 3]
```diff
@@ -64,6 +66,15 @@ while getopts :gi:n:p:x:c:r:sA:P:E: arg; do
...
+    CA_HUB="$(echo ${ALL_HUB_TAG}|cut -f1 -d,)"
+    CA_TAG="$(echo ${ALL_HUB_TAG}|cut -f1 -d,)"
```
Comment (baodongli) [inline install/updateVersion.sh]: ditto

File: install/updateVersion.sh (hunk 1 above)
Comment (sdake) [reply install/updateVersion.sh]: fixed.

File: install/updateVersion.sh (hunk 2 above)
Comment (sdake) [reply install/updateVersion.sh]: fixed.

File: install/updateVersion.sh (hunk 3 above)
Comment (sdake) [reply install/updateVersion.sh]: fixed.

File: DEV-GUIDE.md [hunk 4]
```diff
@@ -246,39 +251,29 @@ tooling functions correctly
...
+### Building and pushing the containers
 
-This tool builds and publishes Mixer container images to the specified
-registry.
+Build the containers in your local docker cache:
 
-```
-bin/publish-docker-images.sh -h gcr.io/my-project -t my-tag
+```shell
+sudo -E make docker
```
Comment (ldemailly) [inline DEV-GUIDE.md]: should not need sudo

File: DEV-GUIDE.md [hunk 5]
```diff
@@ -246,39 +251,29 @@ tooling functions correctly
...
 ```
 
-where
+Push the containers to your registry:
 
-* The `-h` parameter `gcr.io/my-project` is the composition of the registry
-  hostname and the project id. This should be customized.
-* The `-t` parameter `my-tag` is the desired tag. This should be customized.
+```shell
+sudo -E make push
```
Comment (ldemailly) [inline DEV-GUIDE.md]: same, why sudo

File: DEV-GUIDE.md (hunk 4 above)
Comment (sdake) [reply DEV-GUIDE.md]: I agree - sudo is not needed.  I had originally ran this without adding my id to the docker group.  As a result, docker.sock was not available.

File: DEV-GUIDE.md [hunk 6]
```diff
@@ -69,6 +69,15 @@ To run some of Istio's examples and tests, you need to set up Docker server.
 Please follow [these instructions](https://docs.docker.com/engine/installation/)
 for how to do this for your platform.
 
+Ensure your UID is in the docker group to access the docker daemon as a non-root user:
+
+```shell
+sudo adduser username docker
```
Comment (ldemailly) [inline DEV-GUIDE.md]: you could just put $USER or $(id -un)

File: DEV-GUIDE.md [hunk 7]
```diff
@@ -80,6 +89,11 @@ export GOPATH=~/go
 export PATH=$PATH:$GOPATH/bin
 export ISTIO=$GOPATH/src/istio.io # eg. ~/go/src/istio.io
 
+# Please change HUB and TAG to a desired HUB and TAG for custom docker
+# container builds.
+export HUB="gcr.io/yourid"
```
Comment (ldemailly) [inline DEV-GUIDE.md]: probably should be docker.io ? (not everyone is on gcr.io)

File: DEV-GUIDE.md [hunk 8]
```diff
@@ -80,6 +89,11 @@ export GOPATH=~/go
...
+export TAG=`whoami`
```
Comment (ldemailly) [inline DEV-GUIDE.md]: Isn't there a default in the makefile to make a tag with username and date ?

File: DEV-GUIDE.md (hunk 6 above)
Comment (sdake) [reply DEV-GUIDE.md]: ya - I can change this to sudo -E adduser $USER docker ($USER iirc is overridden within a sudo environment)

File: DEV-GUIDE.md (hunk 7 above)
Comment (sdake) [reply DEV-GUIDE.md]: sounds good.

File: DEV-GUIDE.md (hunk 8 above)
Comment (sdake) [reply DEV-GUIDE.md]: Ya.  I'll remove the tag or give greater explanation as to when TAG should be overridden.

File: DEV-GUIDE.md (hunk 6 above)
Comment (ldemailly) [reply DEV-GUIDE.md]: `sudo adduser $USER docker` works because $USER is expanded by the current shell, not by sudo

**Pull request merged**